Assets/Scripts/Screeps3D/Rooms/RoomUnpacker.cs

```diff
@@ -135,10 +140,42 @@ private void UnpackFlags(JSONObject roomData)
                 var dataArray = flagStr.Split('~');
                 if (dataArray.Length < 5)
                     continue;
+                flagList.Add(dataArray[0]);
                 var flag = ObjectManager.Instance.GetFlag(dataArray);
                 flag.FlagDelta(dataArray, _room);
                 _room.Objects[flag.Name] = flag;
             }
+            
+            // process existing flag deltas
+            _removeList.Clear();
+            foreach (var kvp in _room.Objects.Values.OfType<Flag>().ToList())
+            {
+                
+                var id = kvp.Id;
+                var flagObject = kvp;
+
+                if (flagList.Contains(id))
+                {
+                    activeFlagList.Add(id);
+
+                }
+                else if (flagObject.Room != _room)
+                {
+                    _removeList.Add(id);
+                    continue;
+                }
```

Comment:
Ahh, that is why it was there, I was thinking it was to catch some bad data from the API in case something was sent for the wrong room, in the case, I could clean this up.